### PR TITLE
Use non default ports

### PR DESCRIPTION
**Pull request by Koncpa**
Keylime mainly use default ports set in
configuration file. Set non default ports
and check if keylime work on basic
functionality properly.

#130

### CODE REVIEW

File: functional/keylime-non-default-ports/test.sh [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            rlRun "tail -20 $(limeAgentLogfile) | grep 'Executing revocation action local_action_modify_payload'"
+            rlRun "tail $(limeAgentLogfile) | grep 'A node in the network has been compromised: 127.0.0.1'"
+            rlAssertNotExists /var/tmp/test_payload_file
+            cat ${HTTP_SERVER_LOG}
+            rlAssertGrep '\\"type\\": \\"revocation\\", \\"ip\\": \\"127.0.0.1\\", \\"agent_id\\": \\"d432fbb3-d2f1-4a97-9ef7-75bd81c00000\\"' ${HTTP_SERVER_LOG} -i
+            rlAssertNotGrep ERROR ${HTTP_SERVER_LOG} -i
+        fi
+    rlPhaseEnd
+
+    rlPhaseStartTest "Fail keylime agent - setting wrong ports"
```
Comment (kkaarreell) [inline functional/keylime-non-default-ports/test.sh:107]: What is this scenario supposed to do/test?

File: functional/keylime-non-default-ports/test.sh [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        rlRun "limeStopRegistrar"
+        rlRun "limeStopVerifier"
+        rlRun "limeStartVerifier"
+        rlRun "limeWaitForVerifier" 1
+        rlRun "limeStartRegistrar" 
+        rlRun "limeWaitForRegistrar" 1
+    rlPhaseEnd
+
+    rlPhaseStartCleanup "Do the keylime cleanup"
+        rlRun "kill ${HTTP_SERVER_PID}"
```
Comment (kkaarreell) [inline functional/keylime-non-default-ports/test.sh:117]: remove HTTP_SERVER related stuff

File: functional/keylime-non-default-ports/test.sh [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        rlRun "limeUpdateConf tenant verifier_port 18881"
+        rlRun "limeStartVerifier"
+        rlRun "limeWaitForVerifier 18881"
+        rlRun "limeStartRegistrar"
+        rlRun "limeWaitForRegistrar 18890"
+        rlRun "limeStartAgent"
+        rlRun "limeWaitForAgentRegistration ${AGENT_ID}"
+        # create allowlist and excludelist
+        limeCreateTestLists
+        HTTP_SERVER_LOG=$( mktemp )
```
Comment (kkaarreell) [inline functional/keylime-non-default-ports/test.sh:68]: remove HTTP_SERVER and ncat related stuff, it is not necessary for this test

File: functional/keylime-non-default-ports/main.fmf [hunk 4]
```diff
@@ -0,0 +1,27 @@
+summary:  Test of keylime which using non default ports
+description: |
+    Change all default ports to non default ports.
+    Running all services on localhost.
+    Starts verifier, registrar, agent.
+    Add keylime agent.
+    Running verifier and registrar with wrong ports.
```
Comment (kkaarreell) [inline functional/keylime-non-default-ports/main.fmf:7]: Not sure if this last scenario is really useful. What would it prove?

File: functional/keylime-non-default-ports/test.sh [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        rlRun "echo -e '#!/bin/bash\necho boom' > $TESTDIR/keylime-bad-script.sh && chmod a+x $TESTDIR/keylime-bad-script.sh"
+        rlRun "$TESTDIR/keylime-bad-script.sh"
+        rlRun "rlWaitForCmd 'tail \$(limeVerifierLogfile) | grep -q \"Agent $AGENT_ID failed\"' -m 10 -d 1 -t 10"
+        rlRun "limeWaitForAgentStatus $AGENT_ID '(Failed|Invalid Quote)'"
+        if [ -z "$KEYLIME_TEST_DISABLE_REVOCATION" ]; then
+            rlRun "rlWaitForCmd 'tail \$(limeAgentLogfile) | grep -q \"A node in the network has been compromised: 127.0.0.1\"' -m 10 -d 1 -t 10"
+            rlRun "tail -20 $(limeAgentLogfile) | grep 'Executing revocation action local_action_modify_payload'"
+            rlRun "tail $(limeAgentLogfile) | grep 'A node in the network has been compromised: 127.0.0.1'"
+            rlAssertNotExists /var/tmp/test_payload_file
+            cat ${HTTP_SERVER_LOG}
```
Comment (kkaarreell) [inline functional/keylime-non-default-ports/test.sh:101]: remove all HTTP_SERVER related stuff

**Review by kkaarreell — CHANGES_REQUESTED** [review]

File: functional/keylime-non-default-ports/main.fmf [hunk 6]
```diff
@@ -0,0 +1,27 @@
+summary:  Test of keylime which using non default ports
+description: |
+    Change all default ports to non default ports.
```
Comment (kkaarreell) [inline functional/keylime-non-default-ports/main.fmf:3]: Change ports used by keylime services to non default ports ...

**Comment by kkaarreell** [comment]
@Koncpa I am wondering, can we expect this test to fail when running it with keylime SELinux policy? If yes, maybe we would want to tweak it a bit.

**Comment by Koncpa** [comment]
> @Koncpa I am wondering, can we expect this test to fail when running it with keylime SELinux policy? If yes, maybe we would want to tweak it a bit.

I'll try run this test with selinux policy and we'll see.

**Comment by Koncpa** [comment]
> @Koncpa I am wondering, can we expect this test to fail when running it with keylime SELinux policy? If yes, maybe we would want to tweak it a bit.

It seems, that test pass also with keylime SELinux policy. I merge then this PR.

**Review by kkaarreell — APPROVED** [review]

**Pull request merged**